pkg/conversion/mapi2capi/powervs.go

```diff
@@ -0,0 +1,378 @@
+	} else {
+		powerVSMachineSet.Spec.Template.Spec.ClusterName = m.infrastructure.Status.InfrastructureName
+		powerVSMachineSet.Spec.ClusterName = m.infrastructure.Status.InfrastructureName
+	}
+
+	if len(errs) > 0 {
+		return nil, nil, warnings, utilerrors.NewAggregate(errs)
+	}
+
+	return powerVSMachineSet, powerVSMachineTemplate, warnings, nil
+}
+
+// powerVSProviderSpecFromRawExtension unmarshalls a raw extension into an PowerVSMachineProviderSpec type.
+func powerVSProviderSpecFromRawExtension(rawExtension *runtime.RawExtension) (mapiv1.PowerVSMachineProviderConfig, error) {
+	if rawExtension == nil {
+		return mapiv1.PowerVSMachineProviderConfig{}, nil
+	}
+
+	spec := mapiv1.PowerVSMachineProviderConfig{}
+	if err := yaml.Unmarshal(rawExtension.Raw, &spec); err != nil {
+		return mapiv1.PowerVSMachineProviderConfig{}, fmt.Errorf("error unmarshalling providerSpec: %w", err)
+	}
+
+	return spec, nil
+}
+
+// toPowerVSMachine converts PowerVSMachineProviderConfig to IBMPowerVSMachineSpec.
+//
+//nolint:funlen
+func (m *powerVSMachineAndInfra) toPowerVSMachine(providerSpec mapiv1.PowerVSMachineProviderConfig) (*capibmv1.IBMPowerVSMachine, []string, field.ErrorList) {
+	fldPath := field.NewPath("spec", "providerSpec", "value")
+
+	var (
+		errs     field.ErrorList
+		warnings []string
+	)
+
+	serviceInstance, err := convertServiceInstanceToCAPI(fldPath.Child("serviceInstance"), providerSpec.ServiceInstance)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	image, err := convertImageToCAPI(fldPath.Child("image"), providerSpec.Image)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	network, err := convertNetworkToCAPI(fldPath.Child("network"), providerSpec.Network)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	spec := capibmv1.IBMPowerVSMachineSpec{
+		//	ServiceInstanceID: Deprecated, Use ServiceInstance.
+		ServiceInstance: serviceInstance,
+		SSHKey:          providerSpec.KeyPairName,
+		Image:           image,
+		// ImageRef: Not required as image is set above
+		SystemType:    providerSpec.SystemType,
+		ProcessorType: capibmv1.PowerVSProcessorType(providerSpec.ProcessorType),
+		Processors:    providerSpec.Processors,
+		MemoryGiB:     providerSpec.MemoryGiB,
+		Network:       network,
+		// ProviderID. This is populated when this is called in higher level funcs (ToMachine(), ToMachineSet()).
+	}
+
+	if !reflect.DeepEqual(providerSpec.ObjectMeta, metav1.ObjectMeta{}) {
+		// We don't support setting the object metadata in the provider spec.
+		// It's only present for the purpose of the raw extension and doesn't have any functionality.
+		errs = append(errs, field.Invalid(fldPath.Child("metadata"), providerSpec.ObjectMeta, "metadata is not supported"))
+	}
+
+	if len(providerSpec.LoadBalancers) > 0 {
+		// TODO: CAPIBM only applies load balancers to the control plane nodes. We should always reject LBs on non-control plane and work out how to connect the control plane LBs correctly otherwise.
```

Comment:
Need to track this in Jira to follow up on right?

Reply:
Yeah, currently i made a note of it, I will create a Jira as well.

Reply:
Please create the Jira and add the card number to the TODO as we did for AWS, I'd like to see that before we merge this please

Reply:
Done, Created a Jira task and added it in comment.